### PR TITLE
refac: grid area overview - frontend

### PR DESCRIPTION
**Pull request by michaelmajgaard**

### CODE REVIEW

File: libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts [hunk 1]
```diff
@@ -14,50 +14,52 @@
...
-  ],
+  imports: [DhMarketParticipantGridAreaOverviewComponent],
 })
-export class DhGridAreasShellComponent {
-  constructor(private store: DhMarketParticipantGridAreaOverviewDataAccessApiStore) {
-    this.store.init();
+export class DhGridAreasShellComponent implements OnInit, OnDestroy {
+  private readonly gln = new RegExp('^[0-9]+$');
+  private apollo = inject(Apollo);
+  private destroy = new Subject<void>();
```
Comment (FirestarJes) [inline libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts:35]: Should probably use "private destroyRef = inject(DestroyRef);"

File: libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts [hunk 2]
```diff
@@ -14,50 +14,52 @@
...
+    useInitialLoading: true,
+    notifyOnNetworkStatusChange: true,
+    query: GetGridAreaOverviewDocument,
+  });
+
+  isLoading = false;
+  rows: GridAreaOverviewRow[] = [];
+
+  ngOnInit(): void {
+    this.getActorsQuery$.valueChanges.pipe(takeUntil(this.destroy)).subscribe((result) => {
```
Comment (FirestarJes) [inline libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts:47]: Then use ".pipe(takeUntilDestroyed(this.destroyRef)).subscribe"

**Review by FirestarJes — CHANGES_REQUESTED** [review]

File: libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts (hunk 2 above)
Comment (michaelmajgaard) [reply libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts]: fixed

File: libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts (hunk 1 above)
Comment (michaelmajgaard) [reply libs/dh/market-participant/grid-areas/shell/src/lib/dh-grid-areas-shell.component.ts]: fixed

**Review by FirestarJes — APPROVED** [review]

**Review by FirestarJes — APPROVED** [review]

**Pull request merged**